<?xml version="1.0"?>
<launch>
  <node pkg="pointcloud2laser" type="pointcloud2laser" name="pointcloud2laser" output="screen" respawn="false" >
    <!-- See all params at http://wiki.ros.org/pointcloud_to_laserscan -->
      
    <!-- Min and max height to sample from depth data - these values worked for my drone -->
    <param name="min_height" value="0"/>
    <param name="max_height" value="2"/>
      
    <!-- Min and max range range of generated laser scan - set this to match your depth sensor -->
    <param name="range_min" value="0.2"/>
    <param name="range_max" value="5.0"/>

    <param name="transform_tolerance" value="1.0"/>

    <!-- Frame of your depth sensor -->
     <param name="target_frame" value="camera_scan"/>

    <!-- Topic from which to read PointCloud2 -->
    <param name="cloud_topic" value="/camera/depth_registered/points"/>
    
    <!-- Topic to which LaserScan will be published -->
    <param name="scan_topic" value="/camera_scan"/>
  </node>
</launch>
